nav2_graceful_controller/src/graceful_controller.cpp

```diff
@@ -195,21 +195,19 @@ geometry_msgs::msg::TwistStamped GracefulController::computeVelocityCommands(
     // Else, fall through and see if we should follow control law longer
   }
 
-  // Precompute distance to candidate poses
-  std::vector<double> target_distances;
-  computeDistanceAlongPath(transformed_plan.poses, target_distances);
```

Comment:
I believe this will cause a minor regression - the getLookaheadPoint() function simply looks at the distance between the target point and the robot - not the distance ALONG the path. In some instances, this can lead to the controller massively shortcutting the desired path.

I would suggest making getLookaheadPoint use distance along path instead of hypot().

Reply:
Right!
 I can make this change in this PR. 
Or another issue/PR would be appropriate for this?

Reply:
@SakshayMahna I believe this item is resolved, correct? @mikeferguson can you take another look?